src/main/java/models/Deadline.java

```diff
@@ -0,0 +1,59 @@
+package models;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Event class that represents a Deadline that will be saved by Dub.
+ */
+public class Deadline extends Task {
+
+    /** The deadline of the task. */
+    private final LocalDate time;
+
+    /**
+     * Constructor of the Deadline class.
+     *
+     * @param description Description of the Deadline.
+     * @param time Deadline of the task.
+     */
+    public Deadline(String description, LocalDate time) {
+        super(description);
+        this.time = time;
+    }
+
+    /**
+     * Equals function that check whether 2 objects are equal or not.
+     *
+     * @param obj Object that will be compared.
+     * @return True if the object are equal.
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj instanceof Deadline) {
+            Deadline temp = (Deadline) obj;
+            return temp.toString().equals(this.toString());
+        }
+
+        return false;
+    }
+
+    /**
+     * Return string implementation of the Deadline object.
+     *
+     * @return String implementation of the Deadline.
+     */
+    @Override
+    public String toString() {
+        if (this.isDone) {
+            return "[D][X] " + this.description + " (by: "
+                    + time.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+        }
+        return "[D][ ] " + this.description + " (by: "
+                + time.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```

Comment:
Might be repetitive if you return the 'almost' the same string representation. You may want to store a variable called `iconString` which contains `"[X]"` or `"[ ]"` based on `this.isDone`. Then you can concatenate the strings altogether. Seems neater to me this way.